springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,329 @@
+  @ResponseBody
+  public ResponseEntity<String> run(
+      @RequestParam(required = false, defaultValue = DEFAULT_OUTPUT_BASE) String output_base,
+      @RequestParam(required = false, defaultValue = DEFAULT_NUM_THREADS) String threads,
+      @RequestParam(required = false, defaultValue = DEFAULT_COUNTRY_CODE) String country_code,
+      @RequestParam() String url,
+      @RequestParam(required = false, defaultValue = Arguments.VALIDATION_REPORT_NAME_JSON)
+          String validation_report_name,
+      @RequestParam(required = false, defaultValue = Arguments.SYSTEM_ERRORS_REPORT_NAME_JSON)
+          String system_error_report_name,
+      @RequestParam(required = false, defaultValue = "dataset id value") String dataset_id,
+      @RequestParam(required = false, defaultValue = "commit sha value") String commit_sha) {
+
+    Arguments args =
+        queryParametersToArguments(
+            output_base,
+            threads,
+            country_code,
+            url,
+            validation_report_name,
+            system_error_report_name);
+    JsonObject root = new JsonObject();
+    root.add(PROPERTIES_JSON_KEY, new JsonObject());
+    StringBuilder messageBuilder = new StringBuilder();
+    HttpStatus status;
+    final long startNanos = System.nanoTime();
+
+    if (!CliParametersAnalyzer.isValid(args)) {
+      status = HttpStatus.BAD_REQUEST;
+      messageBuilder.append(
+          "%Bad request. Please check query parameters and execution logs for more information.\n");
+      root.getAsJsonObject(PROPERTIES_JSON_KEY)
+          .addProperty(MESSAGE_JSON_KEY, messageBuilder.toString());
+      return new ResponseEntity<>(GSON.toJson(root), status);
+    }
+    GtfsInput gtfsInput;
+    GtfsFeedContainer feedContainer;
+    ValidatorLoader validatorLoader;
+    NoticeContainer noticeContainer = new NoticeContainer();
+    try {
+      validatorLoader = new ValidatorLoader();
+    } catch (ValidatorLoaderException e) {
+      messageBuilder.append(e.getMessage());
+      status = HttpStatus.INTERNAL_SERVER_ERROR;
+      root.getAsJsonObject(PROPERTIES_JSON_KEY)
+          .addProperty(MESSAGE_JSON_KEY, messageBuilder.append(e.getMessage()).toString());
+      return new ResponseEntity<>(GSON.toJson(root), status);
+    }
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+    try {
+      gtfsInput = createGtfsInput(args);
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException);
+      messageBuilder.append(ioException.getMessage());
+      status = HttpStatus.BAD_REQUEST;
+      root.getAsJsonObject(PROPERTIES_JSON_KEY)
+          .addProperty(
+              MESSAGE_JSON_KEY, messageBuilder.append(ioException.getMessage()).toString());
+      exportReport(noticeContainer, args);
+      return new ResponseEntity<>(GSON.toJson(root), status);
+    } catch (URISyntaxException uriSyntaxException) {
+      logger.atSevere().withCause(uriSyntaxException);
+      status = HttpStatus.NOT_FOUND;
```

Comment:
Shouldn't this be `HttpStatus.BAD_REQUEST` as well? It would indicate "if URL is malformed" according to GtfsInput doc and https://docs.oracle.com/javase/8/docs/api/java/net/URISyntaxException.html

Reply:
Indeed! Thanks.

Reply:
`BAD_REQUEST` seems to be appropriate for `IOEception`s. Will modify accordingly.